Formula/g/gnu-cobol.rb

```diff
@@ -20,40 +24,97 @@ class GnuCobol < Formula
     sha256 x86_64_linux:   "4bf957f586d155f1034ac90ac24edff1253e9de935336680689d85f0bbdcf45c"
   end
 
+  head do
+    # there's no offical git mirror, so we use the current upstream SVN repo
+    url "https://svn.code.sf.net/p/gnucobol/code/trunk"
+
+    # tools for bootstrapping
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "flex" => :build
+    depends_on "gettext" => :build
+    depends_on "help2man" => :build
+    depends_on "libtool" => :build
+    depends_on "texinfo" => :build
+
+    # head features multiple indexed handlers
+    depends_on "lmdb"
+    depends_on "unixodbc"
+    # TODO: add "visam" and --with-visam, once formula is added
+
+    # head should be tested before installation
+    uses_from_macos "curl"
+    uses_from_macos "perl"
+  end
+
+  depends_on "pkg-config" => :build
   depends_on "berkeley-db"
   depends_on "gmp"
+  depends_on "json-c"
 
-  def install
-    # both environment variables are needed to be set
-    # the cobol compiler takes these variables for calling cc during its run
-    # if the paths to gmp and bdb are not provided, the run of cobc fails
-    gmp = Formula["gmp"]
-    bdb = Formula["berkeley-db"]
-    ENV.append "CPPFLAGS", "-I#{gmp.opt_include} -I#{bdb.opt_include}"
-    ENV.append "LDFLAGS", "-L#{gmp.opt_lib} -L#{bdb.opt_lib}"
+  uses_from_macos "libxml2"
+  uses_from_macos "ncurses"
 
+  def install
     # Avoid shim references in binaries on Linux.
     ENV["LD"] = "ld" unless OS.mac?
 
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
+    indexed_args = ["--with-db"]
+
+    if build.head?
+      indexed_args << "--with-lmdb"
+      indexed_args << "--with-odbc"
+      indexed_args << "--with-indexed=db"
+
+      # go with whatever gettext we have for head
+      system "sed", "-i", "s/AM_GNU_GETTEXT_VERSION/AM_GNU_GETTEXT_REQUIRE_VERSION/", "configure.ac"
+      # full bootstrap
+      chmod 0755, "build_aux/bootstrap"
+      system "build_aux/bootstrap", "install"
+    end
+
+    system "./configure", *std_configure_args,
                           "--disable-silent-rules",
-                          "--prefix=#{prefix}",
                           "--with-libiconv-prefix=/usr",
-                          "--with-libintl-prefix=/usr"
+                          "--with-libintl-prefix=/usr",
+                          *indexed_args,
+                          "--with-json=json-c",
+                          "--with-xml2",
+                          "--with-curses=ncurses"
+
+    # currently not matching head, so disabled *for now*
+    # system "po/update-linguas.sh" if build.head?
+
+    system "make"
+    if build.head?
+      # don't let the system go unexpected unstable (possibly too much)
+      system "make", "test"
+      system "make", "check"
+    end
```

Comment:
The issue here is that head (and the installed libcob) is not compatible to stable, so if you update a "working" head, then update that after you've compiled your thousands of COBOL sources and find it to be unstable, you cannot just switch to stable (as this would mean you need to recompile every COBOL program) - or can you somehow tell the system to go back to an earlier head version you've installed a week before?

The idea was that people can easily use head with confidence and if they try to update and have a failure they are able to create an upstream bug referencing the tests.

If you tell me that's the design of head "you may break your system and cannot easily undo that", then I'll drop those tests. If it is just "too much" then I could also drop `make test`, which would also drop perl (test executor) and curl (download one additional file; maybe there's a gem that should be used instead in general).
Still `make check` runs quite long (because passing something like `"TESTSUITEFLAGS=--jobs=$(nproc)"` is not used).

Your decision.

Reply:
The only reason HEAD installs still exist in Homebrew is so the developers of the program can build a pre-release version of their software to test it. If something breaks it's up to the user to fix it. The level of support/confidence in HEAD installs you're describing does not and should not exist.

Reply:
> The level of support/confidence in HEAD installs you're describing does not and should not exist.

Thanks for helping me to understand how that is meant. It seems to me that https://docs.brew.sh/Formula-Cookbook#unstable-versions-head should cover that better.

I now dropped that "confidence testing" (which would actually allow a GnuCOBOL developer to easily *do* test) and think this makes the PR "complete".